### PR TITLE
FIX: slow `read_off` method

### PR DESCRIPTION
**Pull request by safesintesi**
Had to use this method for a project and it was incredibly slow.

Fixed changing the engine and tweaking the parameters so that no new import is required.

Also fixes:
- comments in file
- ignores edges that were previously read as faces
- direct type casting instead of casting after creating the `DataFrame`
- coercing error on obj.py
- changed jit decorator to njit
- little spelling mistake :)

I tried to modify the least I could.

Problems not solved:
- OFF files actually can have heterogeneous shapes... :(

### CODE REVIEW

**Comment by safesintesi** [comment]
Not sure how to act if a file exists and respects the standard but has no points and no shapes.
At first I added a check that raises a ValueError, but then I commented it out because returning an empty point cloud might be better.
Thinking about it, maybe a loop that loads a bunch of files shouldn't break just because one of them is empty.
Thoughts about this?

**Comment by daavoo** [comment]
> Not sure how to act if a file exists and respects the standard but has no points and no shapes. At first I added a check that raises a ValueError, but then I commented it out because returning an empty point cloud might be better. Thinking about it, maybe a loop that loads a bunch of files shouldn't break just because one of them is empty. Thoughts about this?

I think it is ok (and would prefer it) to raise an exception. People could handle exceptions in their loop

File: pyntcloud/io/off.py [hunk 1]
```diff
@@ -22,22 +25,38 @@
                 n_faces = int(line[1])
                 break
 
+        # if (n_points == 0):
+            # raise ValueError('The file has no points')
```
[the remark this answers is not in the document]
Comment (safesintesi) [reply pyntcloud/io/off.py]: Removed comments, now it raises a ValueError on empty files

**Comment by safesintesi** [comment]
@daavoo done 🦤

**Comment by safesintesi** [comment]
Kindly pinging to remind someone about this <3

**Comment by tflidd** [comment]
Just on the commit f9bbff7cde3c996b70045dcc729d97c1d64c82a4, this works for me :+1:

File: pyntcloud/io/obj.py [hunk 2]
```diff
@@ -72,7 +72,9 @@ def read_obj(filename):
         for i in range(sum(c.isdigit() for c in f[0].split(" "))):
             mesh_columns.append("v{}".format(i + 1))
 
-    mesh = pd.DataFrame([re.split(r'\D+', x) for x in f], dtype='i4', columns=mesh_columns).astype('i4')
+    # trying to coerce type to integer throws error, casted afetr passes tests
+    # mesh = pd.DataFrame([re.split(r'\D+', x) for x in f], dtype='i4', columns=mesh_columns).astype('i4')
```
Comment (daavoo) [inline pyntcloud/io/obj.py:76]:
```suggestion
```

**Review by daavoo — DISMISSED** [review]

**Review by daavoo — APPROVED** [review]
Thanks @safesintesi

**Pull request merged**